util/dvsim/Runner.py

```diff
@@ -5,49 +5,137 @@
 
 from collections import OrderedDict
 import logging as log
+from signal import SIGINT, signal
+import threading
 import time
 
 from utils import VERBOSE
-from Deploy import Deploy
+from Deploy import Deploy, DeployError
 
 
-class TargetStatus():
-    '''An object to track the status of a run for a given target'''
+class TargetRunner():
+    '''An object to run the jobs for a given target'''
     def __init__(self):
-        self.counters = OrderedDict()
-        self.counters['Q'] = 0
-        self.counters['D'] = 0
-        self.counters['P'] = 0
-        self.counters['F'] = 0
-        self.counters['K'] = 0
-        self.counters['T'] = 0
+        # Sets of items, split up by their current state. The sets are disjoint
+        # and their union equals the keys of self.item_to_status.
+        self._queued = set()
+        self._running = set()
+        self._passed = set()
+        self._failed = set()
+        self._killed = set()
```

Comment:
This probably defeats the purpose of ordering tests by iteration # in `SimCfg._expand_run_list`.

Reply:
Oh, good point. I've changed `_queued` to be a list, which should ensure everything gets popped in order.